packages/mui-material/src/TextField/TextField.test.js

```diff
@@ -244,12 +228,32 @@ describe('<TextField />', () => {
     });
   });
 
-  it('should trigger `onClick` only once', () => {
-    const handleClick = spy();
-    const { getByRole } = render(
-      <TextField variant="outlined" label="Test" onClick={handleClick} />,
-    );
-    fireEvent.click(getByRole('textbox'));
-    expect(handleClick.callCount).to.equal(1);
+  describe('event: click', () => {
+    it('should trigger `onClick` only once', () => {
+      const handleClick = spy();
+      const { getByRole } = render(
+        <TextField variant="outlined" disabled label="Test" onClick={handleClick} />,
+      );
+      fireEvent.click(getByRole('textbox'));
+      expect(handleClick.callCount).to.equal(1);
+    });
+
+    it('registers `onClick` on the root slot', () => {
+      const handleClick = spy();
+      const { getByTestId } = render(
+        <TextField
+          data-testid="root"
+          onClick={handleClick}
+          inputProps={{
+            onClick: handleClick,
+          }}
+        />,
+      );
+
+      const root = getByTestId('root');
+
+      fireEvent.click(root);
+      expect(handleClick.callCount).to.equal(1);
+    });
```

Comment:
This strange looks strange to me. I think the way we test "registers `onClick` on the root slot" is by firing the click on the textbox, adding onClick on the TextField, and checking event.currentTarget.

Reply:
@oliviertassinari Thanks for the suggestion ~ I've changed the test to check that instead

(Also removed the "fires only once" test, it's redundant)